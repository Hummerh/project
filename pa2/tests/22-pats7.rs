enum Either {
   Left(i32),
   Right(bool)
}

fn main() {
   let x = Either::Left(2);
   match (x) {
      Either::Left(n) => { 1; n; 3; },
      Either::Right(n) => { n; }
   }
}
  
