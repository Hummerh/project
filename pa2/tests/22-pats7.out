(crate
    (items
        (enum-def
            (id
                (Either))
            (enum-ctor-defs
                (enum-ctor-def
                    (id
                        (Left))
                    (enum-ctor-params
                        (type-i32)))
                (enum-ctor-def
                    (id
                        (Right))
                    (enum-ctor-params
                        (type-bool)))))
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (x)))
                    (enum
                        (enum-ctor
                            (id
                                (Either))
                            (id
                                (Left)))
                        (enum-ctor-params
                            (lit-dec))))
                (match
                    (id
                        (x))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-enum
                                    (enum-ctor
                                        (id
                                            (Either))
                                        (id
                                            (Left)))
                                    (pat-enum-ctor-params
                                        (pat-id
                                            (id
                                                (n))))))
                            (block
                                (lit-dec)
                                (id
                                    (n))
                                (lit-dec)))
                        (match-arm
                            (pats
                                (pat-enum
                                    (enum-ctor
                                        (id
                                            (Either))
                                        (id
                                            (Right)))
                                    (pat-enum-ctor-params
                                        (pat-id
                                            (id
                                                (n))))))
                            (block
                                (id
                                    (n))))))))))